src/gc.rs

```diff
@@ -133,20 +134,28 @@ impl Collector {
                 // block.
                 let obj = unsafe { Gc::from_raw(ptr as *const i8) };
                 if self.colour(obj) == Colour::Black {
-                    break;
+                    continue;
                 }
-
                 self.mark(obj, Colour::Black);
             }
+
+            // Check each word in the allocation block for pointers.
+            for addr in (ptr..(ptr + size)).step_by(WORD_SIZE) {
```

Comment:
Are the inner brackets `(ptr + size)` necessary? I guess not, but maybe worth checking with `rust fmt`?

Reply:
They are not necessary no. `rust fmt` doesn't care either way. Personally I prefer them as I think it makes it more readable, but I'm not that fussed either way if you think I should remove them.

Reply:
Sure. They're gone now